### PR TITLE
Micky Comments first set

### PR DESCRIPTION
**Pull request by MickyHoltzman**

### CODE REVIEW

File: ExperimentManager1.py [hunk 1]
```diff
@@ -1,43 +1,43 @@
-import time
```
Comment (Anna4142) [inline ExperimentManager1.py:1]: OK

File: ExperimentManager1.py [hunk 2]
```diff
@@ -1,43 +1,43 @@
...
-import time
-from Sound_manager import SoundManager
-from vimba import *
-from VideoAnalyser1 import Video_Analyzer
+# Micky: I suggest to return to the original file name. the '1' suffix is confusing
+
+from VideoAnalyzerSim import VideoAnalyzer
```
Comment (Anna4142) [inline ExperimentManager1.py:3]: Micky please take it out, I want the master to impoer the real video analyzer.

File: simulated_mouse.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import random
-from VideoAnalyzer import Video_Analyzer
-from MouseMonitor1 import MouseMonitor
+#from VideoAnalyzer import Video_Analyzer           # delet these lines, please. They are nor not needed. Still python tries
```
Comment (Anna4142) [inline simulated_mouse.py]: ok

File: ValveControl.py [hunk 4]
```diff
@@ -14,13 +14,16 @@ def OpenValve(self, duration):
...
         self.arduino.DigitalHigh(self.channel)  # Set the pin high
-        print(self.arduino.DigitalHigh(self.channel))
+        #print(self.arduino.DigitalHigh(self.channel))     # Micky: I hope this is not something I did :-)
+                                                          #        why printing the return value of a function that does not return any value?
         self.valveopen = True
 
     def IsValveOpen(self):
         if self.valveopen:
             if time.time() - self.starttime > self.duration:
-                print("delivered")
```
Comment (Anna4142) [inline ValveControl.py:23]: agreed

File: StateManager.py [hunk 5]
```diff
@@ -99,7 +99,7 @@ def SetTimeOut(self, decision_time, return_time):
 
     def DetermineState(self, events):
         TransitionEvents = self.TransitionEvent[self.current_state]
-        print("event",TransitionEvents)
```
Comment (Anna4142) [inline StateManager.py:102]: agreed

File: RewardManager.py [hunk 6]
```diff
@@ -40,8 +45,9 @@ def deliver_reward(self, scenario, mouse_id, reward_time):
         valve_index = self._get_valve_index(scenario, mouse_id)
         if valve_index is not None:
             thread = threading.Thread(target=self._threaded_deliver, args=(valve_index, reward_time))
-            thread.start()
-            thread.join()
+            thread.start()                         # Micky: You are not using the threads correctly. The Start() function
+                                                   #        will return after the valve is closed.
+            thread.join()                          # Micky: I think the join() fucntion is not doing anything.
```
Comment (Anna4142) [inline RewardManager.py:50]: true

File: MainWindow.py [hunk 7]
```diff
@@ -1,5 +1,8 @@
+# Mikcy: The MainWindow file name is a bit of a mis nomenclature. it a left over from the old code where this module actually lanuceched
+#        the main gui window. I know this sounds like a pure nuisance but remember a code is written only once and used (read, maintained, changed)
+#        by generations of programmers. Names are important for understanding. Care to try another name?
 from ExperimentManager1 import ExperimentManager
-from experimentgui import ExperimentGUI, OpponentType
+from experimentgui import ExperimentGUI, OpponentType     # Micky: I do not understand how this works. OpponenType is not defined in experimentGui
```
Comment (Anna4142) [inline MainWindow.py:5]: yes but it inherits it from an import it has

File: ExperimentManager1.py [hunk 8]
```diff
@@ -304,7 +330,8 @@ def start_streaming_exp(self, num_trial, duration, time_decision, opponent_type,
             # print(f"next State: {nextstate}")
             if nextstate != currentstate:
                 currentstate = nextstate
+                print(f"Current State: {nextstate}")           # Micky: A slight change of order and text to improve readability
                 self.StateActivity(currentstate, mouse1sim, mouse2sim)
-                print(f"next State: {nextstate}")
-            if currentstate == States.End:
+
+            if currentstate == States.End:               # Micky: No need to break. Your while loop is using the same condition.
```
Comment (Anna4142) [inline ExperimentManager1.py:336]: ok

File: ExperimentManager1.py [hunk 9]
```diff
@@ -264,9 +288,11 @@ def start_streaming_exp(self, num_trial, duration, time_decision, opponent_type,
 
             else:
                 zone_activations = self.videoAnalyser.process_single_frame()
-                print("zone activations", zone_activations)
+                #print("zone activations", zone_activations)    # Micky: This prints every loop cycle. way to fast to be on any use. it floods the
+                                                                #         the consol window and makes it impossible to see the important data. 
 
-                if opponent_type == OpponentType.MOUSE_MOUSE:
```
Comment (Anna4142) [inline ExperimentManager1.py:269]: ok

File: ExperimentManager1.py [hunk 10]
```diff
@@ -255,7 +277,9 @@ def start_streaming_exp(self, num_trial, duration, time_decision, opponent_type,
         state_history = []
         while currentstate != States.End:
             # print(f"Current State: {currentstate}")
-            state_history.append(currentstate)
```
Comment (Anna4142) [inline ExperimentManager1.py:258]: need it as i might need the state history for the simulated learner

File: ExperimentManager1.py [hunk 11]
```diff
@@ -187,14 +204,19 @@ def StateActivity(self, state, mouse1simulated, mouse2simulated):
...
+            #self.sound_manager.play_sound('beep')            # Micky: Do we need a beep hear?
+            Play(Sounds.Start)
 
         elif state == States.TrialCompleted:
             # Increment the trial number counter
             self.numcompletedtrial += 1
             print("Trial Completed. Number of completed trials: ", self.numcompletedtrial)
             # self.trial_logger.log_trial_data(self.numcompletedtrial, "Completed Trial", self.opponent_choice, self.mouse_choice, self.mouse_reward, self.opponent_reward)
-
+                                                        # Micky: Why are the logger writes, commented out? There is no need to.
```
Comment (Anna4142) [inline ExperimentManager1.py:215]: have corrected in latest version

File: ExperimentManager1.py [hunk 12]
```diff
@@ -48,14 +48,14 @@ def __init__(self):
...
-        # Initialize CSV file and video capture
+        # Initialize CSV file and video capture       # Micky: I think it is time to delete these lines. You moved them to the relevant classes
         # self.initialize_csv_logging()
         # self.initialize_video_capture()
 
         ##
         ### VARIABLES FOR LOG FILE INITIALLY INITIALIZED TO FALSE
-        self.data_file_loc = 'path_to_csv_file.csv'
-        self.cc_var = False
+        self.data_file_loc = 'path_to_csv_file.csv'   # Micky: Use relative file name. it will make the SW more portable
```
Comment (Anna4142) [inline ExperimentManager1.py:57]: yes

File: ExperimentManager1.py [hunk 13]
```diff
@@ -1,43 +1,43 @@
...
 from logger import TrialLogger
 from OpponentType import OpponentType
 
 
-# from ArduinoDigitalSim import ArduinoDigital  ##Anushka-new class
 class ExperimentManager:
     def __init__(self):
         # Initialize other components here
-        self.root = tk.Tk()  # Create a new Tkinter root window if not provided
+        #self.root = tk.Tk()  # Create a new Tkinter root window if not provided   #Micky: The video Analyzer stub should create its own TK root
```
Comment (Anna4142) [inline ExperimentManager1.py:26]: made changes in latest version

File: VideoAnalyzerSim.py [hunk 14]
```diff
@@ -0,0 +1,34 @@
+from pynput.keyboard import Key, Listener
```
Comment (Anna4142) [inline VideoAnalyzerSim.py:1]: prefer the stub

File: experimentgui.py [hunk 15]
```diff
@@ -1,3 +1,10 @@
+# Micky: The label still says trial duration. Shouldn't it be Center Time or something similar?
+# Micky: There is still a need to close the gui window to start the trial, I think it is redundant.
```
Comment (Anna4142) [inline experimentgui.py:2]: changed in latest version

File: ExperimentManager1.py [hunk 16]
```diff
@@ -1,43 +1,43 @@
...
-# from VideoAnalyzerStub import Video_Analyzer
 from MouseMonitor1 import Locations
-from logger import TrialLogger
+from SoundManager import *
 from simulated_mouse import Simulated_mouse
-from ArduinoDigital import ArduinoDigital
 from StateManager import StateManager
 from StateManager import States
 from StateManager import Events
-import experimentgui
```
Comment (Anna4142) [inline ExperimentManager1.py:17]: ok

File: ExperimentManager1.py (hunk 2 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:3]: We need to decide on the name. VideoAnalyzer or Video_Analyzer

File: ExperimentManager1.py (hunk 13 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:26]: OK. Let's keep the comment until I see the changes.

File: ExperimentManager1.py (hunk 12 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:57]: I am keeping the comments until I see the change

File: ExperimentManager1.py (hunk 11 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:215]: OK. I am keeping the comment for the time being.

File: ExperimentManager1.py (hunk 10 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:258]: I do not understand why you need it. Let's discuss. You have the same information in the experiment log file. Also, You do not need the array to be appended every cycle of the loop. Move it to state changes if you really need it.

File: ExperimentManager1.py (hunk 8 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:336]: I made the code change.

File: MainWindow.py (hunk 7 above)
Comment (MickyHoltzman) [reply MainWindow.py:5]: OK. Got it :-)

File: RewardManager.py (hunk 6 above)
Comment (MickyHoltzman) [reply RewardManager.py:50]: So ? How do we fix it?

File: VideoAnalyzerSim.py (hunk 14 above)
Comment (MickyHoltzman) [reply VideoAnalyzerSim.py:1]: OK. You got it.

File: simulated_mouse.py (hunk 3 above)
Comment (MickyHoltzman) [reply simulated_mouse.py]: Done.

File: ExperimentManager1.py (hunk 10 above)
Comment (Anna4142) [reply ExperimentManager1.py:258]: can you keep this line

**Review by Anna4142 — COMMENTED** [review]
reviewed

File: ExperimentManager1.py (hunk 10 above)
Comment (MickyHoltzman) [reply ExperimentManager1.py:258]: Its still there don't worry

**Review by Anna4142 — APPROVED** [review]

**Pull request merged**